contents/testing/writing-integration-test.md

```diff
@@ -0,0 +1,143 @@
+1. Repeat testing until all integration tests are successful.
+
+### Integration Test Plan
+Before the actual implementation of your integration tests, it is important to decide on the approach used. Using a concrete example where by you are designing a online web ordering system. Your application may be split into different modules as shown below:
+
+![Types of integration test plan](integration-test/integration-test-plan-types.png "Types of integration test plan")
+
+As shown, there are a few different approaches of integration testing in which you can adopt depending on the development progress of your application.
+
+#### [Big bang approach](https://www.tutorialspoint.com/software_testing_dictionary/big_bang_testing.html): 
+ 
+This approach involves integrating all the components in your design diagram together and test everything at once in a complete state. This is convenient but it is difficult to isolate defects and raises high chances of missing critical defects.
+
+It is usually used for smaller applications with a few components.
+
+#### [Incremental testing approach](https://www.quora.com/What-is-incremental-testing-in-software):
+
+This approach involves integrating two or more logically related components. The other related components are added and tested for proper functioning. Repeat these steps until all the components are joined and tested successfully.
+
+It is usually preferred with any bigger applications with more than 10 different components. It is further split into the 3 approaches shown below.
+
+| | [Top-down approach](https://www.guru99.com/integration-testing.html#9) | [Bottom-up approach](https://www.guru99.com/integration-testing.html#8) | [Hybrid / sandwich approach](https://www.guru99.com/integration-testing.html#10) |
+| -- | -- | -- | -- |
+| **Description**| Higher level modules are tested with lower level modules until all modules are tested. | Lower level modules are tested with higher level modules until all modules are tested. | A mix of both top-down bottom-up approaches. |
+| **Advantages** | Early discovery of high level architecture / design defects | Easier to create test cases bottom up | Beneficial for big project to distribute tasks on testing
+| | Main control points of the system are tested early | Critical modules on functionalities ae tested first | Allow top-down and bottom-up approach to run side by side | 
+| **Disadvantages** | Significant low level moduls are tested late in the cycle | There is no working system until the last module is build | It is difficult to test for highly interconnected modules |
+| | A [stub](https://stackoverflow.com/questions/463278/what-is-a-stub) is not perfect to simulate data flow | A [driver](http://www.professionalqa.com/test-driver) test is even harder to write than stub | Higher cost from using both driver and stub. You can understand the difference [here](https://www.quora.com/What-is-the-difference-between-stubs-and-drivers-in-software-testing)
```

Comment:
Similar to above, I think you might want to provide high level descriptions of jargon directly in the article itself, so users do not need to go elsewhere to find the information.